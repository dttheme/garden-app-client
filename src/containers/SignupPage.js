import React, { Component } from 'react';

import SignupForm from "../components/SignupForm"
// import './SignupForm.css';

class SignupPage extends Component {
  render() {
    return (
      <div className="signupWrapper">
        <h2>Sign up</h2>
        <SignupForm />
      </div>

    );
  }
}

export default SignupPage;
