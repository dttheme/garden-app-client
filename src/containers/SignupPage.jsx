import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import { createUser } from '../actions/index.actions';


// import './SignupForm.css';

class SignupPage extends Component {
  submitHandler(val) {
    const email = val.email;
    const password = val.password;
    const firstName = val.firstName;
    this.props.createUser(firstName, email, password);
  }
  render() {
    return (
      <div className="signupWrapper w-90 w-60-ns mt5 ml7-ns ml5-m ba-ns bw2-ns pl6-ns pl4-m pl4 pb4">
        <h2 className="f2 tracked">
          Sign up
        </h2>
          <form onSubmit={this.props.handleSubmit(this.submitHandler.bind(this))} method="POST">
            <label
              htmlFor="firstName"
              className="tracked"
              >
              First Name
            </label>
            <Field
              name="firstName"
              component="input"
              type="text"
              placeholder="Bud"
              className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
              required
            />
            <label
              htmlFor="email"
              className="tracked"
              >
              Email
            </label>
            <Field
              name="email"
              component="input"
              type="email"
              placeholder="sprout@email.com"
              className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
              required
            />
            <label
              htmlFor="password"
              className="tracked"
              >
              Password
            </label>
            <Field
              name="password"
              component="input"
              type="password"
              className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
              required
            />
            <button
              type="submit"
              className="w-100 w-70-m w-50-ns f4 link dim ba bw2 ph3 pv2 mt3 mb2 dib near-black"
              >
              Sign up
            </button>
          </form>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({

});

export default reduxForm({
  form: 'signup'
})(connect(mapStateToProps, {createUser})(SignupPage))
