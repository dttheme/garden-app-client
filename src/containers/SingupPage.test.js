import React from 'react';
import { shallow, mount } from 'enzyme';
import { SignupPage } from './SignupPage';

describe('<SignupPage />', () => {

  it('should render page', () => {
    const wrapper = shallow(<SignupPage handleSubmit={() => {}} />);
    expect(wrapper).toMatchSnapshot()
  });

});
