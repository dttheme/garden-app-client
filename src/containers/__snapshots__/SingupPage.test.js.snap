// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupPage /> should render page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignupPage
    handleSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="f2 tracked"
>
          Sign up
</h2>,
        <form
          method="POST"
          onSubmit={undefined}
>
          <label
                    className="tracked"
                    htmlFor="firstName"
          >
                    First Name
          </label>
          <Field
                    className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
                    component="input"
                    name="firstName"
                    placeholder="Bud"
                    required={true}
                    type="text"
          />
          <label
                    className="tracked"
                    htmlFor="email"
          >
                    Email
          </label>
          <Field
                    className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
                    component="input"
                    name="email"
                    placeholder="sprout@email.com"
                    required={true}
                    type="email"
          />
          <label
                    className="tracked"
                    htmlFor="password"
          >
                    Password
          </label>
          <Field
                    className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
                    component="input"
                    name="password"
                    required={true}
                    type="password"
          />
          <button
                    className="w-100 w-70-m w-50-ns f4 link dim ba bw2 ph3 pv2 mt3 mb2 dib near-black"
                    type="submit"
          >
                    Sign up
          </button>
</form>,
      ],
      "className": "dark-gray bg-dark-green w-90 w-80-m w-60-ns mt7-m mt6-ns ml7-ns ml5-m ba-ns bw3-ns b--green pl6-ns pl4-m pl4 pb4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign up",
          "className": "f2 tracked",
        },
        "ref": null,
        "rendered": "Sign up",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="tracked"
              htmlFor="firstName"
>
              First Name
</label>,
            <Field
              className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
              component="input"
              name="firstName"
              placeholder="Bud"
              required={true}
              type="text"
/>,
            <label
              className="tracked"
              htmlFor="email"
>
              Email
</label>,
            <Field
              className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
              component="input"
              name="email"
              placeholder="sprout@email.com"
              required={true}
              type="email"
/>,
            <label
              className="tracked"
              htmlFor="password"
>
              Password
</label>,
            <Field
              className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
              component="input"
              name="password"
              required={true}
              type="password"
/>,
            <button
              className="w-100 w-70-m w-50-ns f4 link dim ba bw2 ph3 pv2 mt3 mb2 dib near-black"
              type="submit"
>
              Sign up
</button>,
          ],
          "method": "POST",
          "onSubmit": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "First Name",
              "className": "tracked",
              "htmlFor": "firstName",
            },
            "ref": null,
            "rendered": "First Name",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "db mb4 mt2 w-100 w-70-m w-50-ns pa2",
              "component": "input",
              "name": "firstName",
              "placeholder": "Bud",
              "required": true,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Email",
              "className": "tracked",
              "htmlFor": "email",
            },
            "ref": null,
            "rendered": "Email",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "db mb4 mt2 w-100 w-70-m w-50-ns pa2",
              "component": "input",
              "name": "email",
              "placeholder": "sprout@email.com",
              "required": true,
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Password",
              "className": "tracked",
              "htmlFor": "password",
            },
            "ref": null,
            "rendered": "Password",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "db mb4 mt2 w-100 w-70-m w-50-ns pa2",
              "component": "input",
              "name": "password",
              "required": true,
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign up",
              "className": "w-100 w-70-m w-50-ns f4 link dim ba bw2 ph3 pv2 mt3 mb2 dib near-black",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign up",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="f2 tracked"
>
            Sign up
</h2>,
          <form
            method="POST"
            onSubmit={undefined}
>
            <label
                        className="tracked"
                        htmlFor="firstName"
            >
                        First Name
            </label>
            <Field
                        className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
                        component="input"
                        name="firstName"
                        placeholder="Bud"
                        required={true}
                        type="text"
            />
            <label
                        className="tracked"
                        htmlFor="email"
            >
                        Email
            </label>
            <Field
                        className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
                        component="input"
                        name="email"
                        placeholder="sprout@email.com"
                        required={true}
                        type="email"
            />
            <label
                        className="tracked"
                        htmlFor="password"
            >
                        Password
            </label>
            <Field
                        className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
                        component="input"
                        name="password"
                        required={true}
                        type="password"
            />
            <button
                        className="w-100 w-70-m w-50-ns f4 link dim ba bw2 ph3 pv2 mt3 mb2 dib near-black"
                        type="submit"
            >
                        Sign up
            </button>
</form>,
        ],
        "className": "dark-gray bg-dark-green w-90 w-80-m w-60-ns mt7-m mt6-ns ml7-ns ml5-m ba-ns bw3-ns b--green pl6-ns pl4-m pl4 pb4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign up",
            "className": "f2 tracked",
          },
          "ref": null,
          "rendered": "Sign up",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="tracked"
                htmlFor="firstName"
>
                First Name
</label>,
              <Field
                className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
                component="input"
                name="firstName"
                placeholder="Bud"
                required={true}
                type="text"
/>,
              <label
                className="tracked"
                htmlFor="email"
>
                Email
</label>,
              <Field
                className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
                component="input"
                name="email"
                placeholder="sprout@email.com"
                required={true}
                type="email"
/>,
              <label
                className="tracked"
                htmlFor="password"
>
                Password
</label>,
              <Field
                className="db mb4 mt2 w-100 w-70-m w-50-ns pa2"
                component="input"
                name="password"
                required={true}
                type="password"
/>,
              <button
                className="w-100 w-70-m w-50-ns f4 link dim ba bw2 ph3 pv2 mt3 mb2 dib near-black"
                type="submit"
>
                Sign up
</button>,
            ],
            "method": "POST",
            "onSubmit": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "First Name",
                "className": "tracked",
                "htmlFor": "firstName",
              },
              "ref": null,
              "rendered": "First Name",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "db mb4 mt2 w-100 w-70-m w-50-ns pa2",
                "component": "input",
                "name": "firstName",
                "placeholder": "Bud",
                "required": true,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email",
                "className": "tracked",
                "htmlFor": "email",
              },
              "ref": null,
              "rendered": "Email",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "db mb4 mt2 w-100 w-70-m w-50-ns pa2",
                "component": "input",
                "name": "email",
                "placeholder": "sprout@email.com",
                "required": true,
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Password",
                "className": "tracked",
                "htmlFor": "password",
              },
              "ref": null,
              "rendered": "Password",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "db mb4 mt2 w-100 w-70-m w-50-ns pa2",
                "component": "input",
                "name": "password",
                "required": true,
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign up",
                "className": "w-100 w-70-m w-50-ns f4 link dim ba bw2 ph3 pv2 mt3 mb2 dib near-black",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign up",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
